packages/forms/test/reactive_integration_spec.ts

```diff
@@ -4358,6 +4358,369 @@ const ValueAccessorB = createControlValueAccessor('[cva-b]');
           fixture.destroy();
         }).not.toThrow();
       });
+
+      describe('unsubscribe from async validator', () => {
+        @Directive({
+          selector: '[app-async-validator]',
+          providers: [{
+            provide: NG_ASYNC_VALIDATORS,
+            multi: true,
+            useExisting: forwardRef(() => AppAsyncValidator)
+          }]
+        })
+        class AppAsyncValidator implements AsyncValidator {
+          unsubscribeCalledTimes = 0;
+
+          validate() {
+            return new Observable<null>(() => {
+              // Note that we intentionally do not call `observer.next` and `observer.complete`
+              // so the subscription will be in non-closed status.
+              return () => {
+                this.unsubscribeCalledTimes++;
+              };
+            });
+          }
+        }
```

Comment:
I think it'd be a bit more easier to work with tests if we log validation start and cancel operations (and cleanup logger during and after each test as needed):
```suggestion
        const logger: string[] = [];
        // ...
        class AppAsyncValidator implements AsyncValidator {
          validate() {
            logger.push('validation started');
            return new Observable<null>(() => {
              // Note that we intentionally do not call `observer.next` and `observer.complete`
              // so the subscription will be in non-closed status.
              return () => {
                logger.push('validation cancelled');
              };
            });
          }
        }
```